### PR TITLE
Navigation bar

### LINKED ISSUE #41: Add Favlist, Loing/Signup/Logout in Navigation bar

### PR DESCRIPTION
**Pull request by SachiGoto**
### Feature Description

Worked on the header. The navigation items changes based on whether a use is logged in or not. 
Login and out status is manually implemented for now. 

Screenshot 2023-08-03 at 4 22 42 PM
User Logged in 

Screenshot 2023-08-03 at 4 22 13 PM
User not logged in 

### Related Issue
closes #41

### Something you want to mention
I was trying to make the most out of Mantine Navigation bar but there is a tiny issue and I feel like I made things unnecessarily complicated. 
Whenever I close the overlay navigation bar, I unexpectedly see the navigation bar with the CSS intended for desktop size.

 

### Checklist before requesting a review

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by Lada496** [comment]
@SachiGoto I'm not 100% sure what's the issue as the video file seems broken but I think it appears because `open` state turns to be `false` right after you click the close button.
So using viewport size can be a solution.
```javascript
const { width } = useViewportSize();
const isMobile = width < 768 // https://v4.mantine.dev/theming/responsive/

const items = LINKS.map((link) =>
    isMobile ? (
      <Link
        key={link.label}
        href={link.link}
        className={cx(classes.link, {
          [classes.linkActive]: activePathname === link.link,
        })}
      >
        {link.label}
      </Link>
    ) : (
      <Link
        key={link.label}
        href={link.link}
      >
        <Button buttonType={link.buttonType} style={{ marginTop: "5%" }}>
          {link.label}{" "}
        </Button>
      </Link>
    )
  );

```
[useViewportSize](https://mantine.dev/hooks/use-viewport-size/)

Another solution is using [MediaQuery](https://mantine.dev/core/media-query/) component

I didn't test this but code will be something like this
```javascript
const items = LINKS.map((link) => <>
<MediaQuery largerThan="sm">
<Link
        key={link.label}
        href={link.link}
      >
        <Button buttonType={link.buttonType} style={{ marginTop: "5%" }}>
          {link.label}{" "}
        </Button>
      </Link>
</MediaQuery>
<MediaQuery smallerThan="sm">
<Link
        key={link.label}
        href={link.link}
        className={cx(classes.link, {
          [classes.linkActive]: activePathname === link.link,
        })}
      >
        {link.label}
      </Link>
</MediaQuery>
</>


```
Please take a loot at these component/hook and try it🤞

File: src/app/components/Header/index.tsx [hunk 1]
```diff
@@ -14,53 +14,91 @@ import {
...
   return (
     <MantineHeader height={HEADER_HEIGHT} className={classes.root}>
       <Container className={classes.header}>
-        <div>
-          <Image src="/images/logo.png" alt="logo" width={45.5} height={57.5} />{" "}
-          Tech Meets
-        </div>
-        <Group spacing={5} className={classes.links}>
-          {items}
+        <Link href="/">
```
Comment (Lada496) [inline src/app/components/Header/index.tsx:75]: I like this link!

**Comment by Lada496** [comment]
Here is some discussion I want to talk with all.
I think it helps users to have consistent stylings if the functionality of buttons are the same. 
In this case, map, fav events, login, signup are all links while logout is a button to logout.
For example, sign up, logout, find events has the same design but their functionalities are different, which might get users confused.
Also I think it's helpful to tell the current location to users.
Let me know how you all think about it.

ref: https://webflow.com/blog/navigation-bar-design

**Comment by SachiGoto** [comment]
@Lada496 Thank you so much for the solution to the issue! I should have set different style based on screen size. Why didn't I think of that before!! 

Also, I don't know why I decided to use Buttons for all the nav items for screen sizes bigger than mobile. You raised a really good point. Maybe I will change header to look like the one below?? 
Screenshot 2023-08-04 at 3 28 43 PM

**Comment by SachiGoto** [comment]
Changed buttons to text ! 

<table>
  <tr>
    <td valign="top">
Screenshot 2023-08-04 at 10 08 02 PM</td>
    <td valign="top">Screenshot 2023-08-04 at 10 07 34 PM</td>
  </tr>
  <tr>
    <td valign="top">

Screenshot 2023-08-04 at 10 07 51 PM</td>
    <td valign="top">Screenshot 2023-08-04 at 10 07 22 PM</td>
  </tr>
</table>

**Comment by keento0809** [comment]
@Lada496 @SachiGoto Actually, I've never considered the design of navbar from the usability perspective before, lol

Thank you for suggesting us a thoughtful point of view and I feel like Sachi's modification (changed buttons to text) sounds nice 👍

**Review by Lada496 — APPROVED** [review]
Thank you @SachiGoto!

**Review by devjoylee — APPROVED** [review]
LGTM 🩷

**Pull request merged**